### PR TITLE
feat:websoket-gateway

### PR DESCRIPTION
**Pull request by romansharapov19**
Prepare the base communication with ui and API including real-time messages and loading the messages from the database.
Add next data refresh on mutation

### CODE REVIEW

File: packages/api/src/chats/chats.gateway.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  @SubscribeMessage('leaveRoom')
+  async handleLeaveRoom(
+    @MessageBody('data') data: { roomId: string },
+    @ConnectedSocket() client: Socket
+  ) {
+    client.leave(data.roomId);
+  }
+
+  @SubscribeMessage('message')
+  async handleConnectedToRoom(
```
Comment (comoser) [inline packages/api/src/chats/chats.gateway.ts:49]:
```suggestion
  async handleMessageToRoom(
```

File: packages/api/src/chats/chats.module.ts [hunk 2]
```diff
@@ -24,6 +25,7 @@ import { Module } from '@nestjs/common';
     FindChatMessageHistoryByRoomIdUsecase,
     CreateChatForRoomUsecase,
     AddMessageToChatUsecase,
+    ChatsGateway,
```
Comment (comoser) [inline packages/api/src/chats/chats.module.ts:28]:
```suggestion
    ChatSocketGateway,
```

File: packages/api/src/rooms/usecases/find-room.usecase.ts [hunk 3]
```diff
@@ -8,11 +8,11 @@ import { RoomsRepository } from '@/rooms/rooms.repository';
 import { Injectable } from '@nestjs/common';
 
 @Injectable()
-export class FindRoomByNameUsecase implements Usecase {
+export class FindRoomUsecase implements Usecase {
```
Comment (comoser) [inline packages/api/src/rooms/usecases/find-room.usecase.ts:11]: why the renames? the usecase was supposed to be specific

File: packages/contract/chats/socket-room-create.request.dto.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export interface SocketRoomCreateRequestDto {
```
Comment (comoser) [inline packages/contract/chats/socket-room-create.request.dto.ts:1]:
```suggestion
export interface SocketCreateRoomRequestDto {
```

File: packages/web-ui/app/rooms/[room]/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+const getRoom = async (roomId: string) => {
+  try {
+    const nextCookies = cookies();
+    const clerkJwtToken = nextCookies.get('__session');
+    const { sessionId } = auth();
+    const res = await apiClient({
+      url: Endpoints.rooms.getRoomById(roomId),
+      options: { method: 'GET', cache: 'no-store' },
+      sessionId: sessionId ? sessionId : '',
```
Comment (comoser) [inline packages/web-ui/app/rooms/[room]/layout.tsx:21]:
```suggestion
      sessionId: sessionId ?? '',
```

File: packages/web-ui/app/rooms/[room]/layout.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      jwtToken: clerkJwtToken ? clerkJwtToken.value : '',
```
Comment (comoser) [inline packages/web-ui/app/rooms/[room]/layout.tsx:22]:
```suggestion
      jwtToken: clerkJwtToken?.value ?? '',
```

File: packages/web-ui/app/rooms/layout.tsx [hunk 7]
```diff
@@ -2,10 +2,10 @@ import '@/styles/globals.css';
 import { cookies } from 'next/headers';
 import { apiClient } from '@/api/apiClient';
 import Endpoints from '@/api/endpoints';
-import { RoomResponseDto } from '@/contract/rooms/room.response.dto.d';
 import { auth } from '@clerk/nextjs';
 
 import { Rooms } from '@/components/rooms';
+import { StateWrapper } from '@/components/state-wrapper';
```
Comment (comoser) [inline packages/web-ui/app/rooms/layout.tsx:8]:
```suggestion
import { StateProvider } from '@/components/state-provider';
```

**Review by comoser — APPROVED** [review]

File: packages/api/src/rooms/usecases/find-room.usecase.ts (hunk 3 above)
Comment (romansharapov19) [reply packages/api/src/rooms/usecases/find-room.usecase.ts:11]: At the moment we dont need to find a room by name so the `FindRoomByNameUsecase` was refactored to find room by ID and renamed to `FindRoomUsecase`

Comment (comoser) [reply packages/api/src/rooms/usecases/find-room.usecase.ts:11]: ok then 👍

**Pull request merged**